File: src/main/java/com/social/meow/model/Post.java
package com.social.meow.model;

import jakarta.persistence.*;

import java.io.Serializable;

@Entity
@Table(name = "posts")
public class Post implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;

    @Column(name = "text")
    private String text;

    @Column(name = "user_id")
    private long userId;

    public Post() {

    }

    public Post(String text, long userId) {
        this.text = text;
        this.userId = userId;
    }

    public long getId() {
        return id;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public long getUserId() {
        return userId;
    }

    public void setUserId(Long userId) {
        this.userId = userId;
    }


    @Override
    public String toString() {
        return "Post [id=" + id + ", text=" + text + ", userId=" + userId + "]";
    }
}

File: src/main/kotlin/com/social/meowkotlin/model/Post.kt
package com.social.meowkotlin.model

import jakarta.persistence.*
import java.io.Serializable

@Entity
@Table(name = "posts")
class Post : Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    val id: Long = 0

    @Column(name = "text")
    var text: String? = null

    @Column(name = "user_id")
    var userId: Long = 0

    constructor()

    constructor(text: String?, userId: Long) {
        this.text = text
        this.userId = userId
    }

    override fun toString(): String {
        return "Post [id=$id, text=$text, userId=$userId]"
    }
}
